react/week4/my-app/src/Context.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from "react";
+
+const Context = React.createContext();
+
+function ContextProvider({ children }) {
+  const [userInput, setUserInput] = useState("");
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  useEffect(() => {
+    if (!userInput || error) {
+      setLoading(false);
+    } else {
+      setLoading(true);
+      fetchGitUsers(userInput);
+    }
+  }, [userInput, error]);
+  const fetchGitUsers = async (input) => {
+    try {
+      await fetch(`https://api.github.com/search/users?q=${input}`)
+        .then((response) => {
+          if (!response.ok) {
+            throw Error("Could not fetch the data ");
+          } else {
+            return response.json();
+          }
```

Comment:
This is a very small thing but I personally prefer to have (if possible) the if/else reversed, meaning `if (response.ok) { ... } else { ... }`.

The idea is to avoid `!` (not), if possible.